src/DGtal/geometry/volumes/BoundedLatticePolytope.h

```diff
@@ -0,0 +1,878 @@
+     *
+     * @param l any list of d+1 points in general positions.
+     * @pre Note that the list must contain Space::dimension+1 points
+     * in general position.
+     */
+    BoundedLatticePolytope( std::initializer_list<Point> l );
+    
+    /**
+     * Constructs the polytope from a simplex given as a range
+     * [itB,itE) of lattice points.  
+     *
+     * @pre Note that the range must contain Space::dimension+1 points
+     * in general position. Otherwise the build polytope is empty.
+     *
+     * @tparam PointIterator any model of forward iterator on Point.
+     * @param itB the start of the range of n+1 points defining the simplex.
+     * @param itE past the end the range of n+1 points defining the simplex.
+     */
+    template <typename PointIterator>
+    BoundedLatticePolytope( PointIterator itB, PointIterator itE );
+
+    /**
+     * Constructs a polytope from a domain and a collection of half-spaces.
+     *
+     * @tparam HalfSpaceIterator any model of forward iterator on HalfSpace.
+     * @param domain a bounded lattice domain.
+     * @param itB the start of the range of half-spaces.
+     * @param itE past the end of the range of half-spaces.
+     *
+     * @param valid_edge_constraints when 'true', tells that there are
+     * half-spaces that represents th constraints on edges (n-2 cells)
+     * lying between two faces (n-1 cells) pointing to different
+     * orthants.
+     */
+    template <typename HalfSpaceIterator>
+    BoundedLatticePolytope( const Domain& domain,
+			    HalfSpaceIterator itB, HalfSpaceIterator itE,
+                            bool valid_edge_constraints = false );
+
+    /**
+     * Initializes a polytope from a domain and a vector of half-spaces.
+     *
+     * @tparam HalfSpaceIterator any model of forward iterator on HalfSpace.
+     * @param domain a bounded lattice domain.
+     * @param itB the start of the range of half-spaces.
+     * @param itE past the end of the range of half-spaces.
+     *
+     * @param valid_edge_constraints when 'true', tells that there are
+     * half-spaces that represents th constraints on edges (n-2 cells)
+     * lying between two faces (n-1 cells) pointing to different
+     * orthants.
+     */
+    template <typename HalfSpaceIterator>
+    void init( const Domain& domain,
+	       HalfSpaceIterator itB, HalfSpaceIterator itE,
+               bool valid_edge_constraints = false );
+
+    
+    /**
+     * Initializes the polytope from a simplex given as a range [itB,itE) of points.
+     *
+     * @param itB the start of the range of n+1 points defining the simplex.
+     * @param itE past the end the range of n+1 points defining the simplex.
+     *
+     * @return 'true' if [itB,itE) was a valid simplex, otherwise
+     * return 'false' and the polytope is empty.
+     *
+     * @note Use DGtal SimpleMatrix::determinant which is not efficient when the
+     * dimension is high. Does not use Eigen to avoid dependency.
+     */
```

Comment:
I'm fine with that but we may already have a facade on SimpleMatrix and Eigen that can be used to abstract the det computation (when eigen is activated). I'd need to double-check in the DEC section

Reply:
As for now, the general code will work only dimension 2 and 3 if you wish to do Minkowski sums. That's why it is not so important.